meeting-notes/2018/2018-10-29--go-core-dev-team-weekly.md

```diff
@@ -0,0 +1,189 @@
+# 💫Golang Core Dev Team Weekly Sync 🙌🏽 October 29, 2018
+
+- **Meeting Lead:** @eingenito
+- **Notetaker:**
+- **Attendees:**
+  - bigs
+  - erikingenito
+  - hannah howard
+  - warpfork
+  - djdv
+  - Adin Schmahmann
+  - Molly Mackinlay
+  - David Dias
+  - Kevin Atkinson
+  - Hector Sanjuan
+  - Travis Person
+- **Recording:** _add link to recording once it's online_
+
+## Agenda
+
+- Ask everyone to put their name into the list of attendees
+- Round of updates
+  - What have you accomplished since the last Weekly?
+  - Were there any blockers? If so, which ones? Is it still blocked? Why?
+  - What is the next important thing you should focus on?
+- Ask for general questions. Could be things like:
+  - I'm stuck with something, I don't know who to ask. Who knows who to ask?
+  - Who can help me with xyz?
+- Plan this week
+  - Select issues to work on
+- Review remaining issues if there is time left
+
+
+## Notes
+
+### Week Update
+
+@name
+ - Done:
+ - Blocked:
+ - Next:
+
+
+@marten-seemann
+
+- Done: completed the switch to tls-tris for QUIC (still a bunch of issues to fix though)
+- Blocked: nothing
+- Next: preparing for the IETF meeting next week
+
+@schomatis (not attending)
+- Done:
+  - Solved some issues and reviewed PRs on UnixFS and related components.
+  - Reviewed more UnixFS/MFS code.
+- Next: Coordinate with Rob to start producing some IPFS Files documentation.
+
+@warpfork
+ - Done:
+   - a few hours on refmt obj traversal improvements.
+   - researching prior art for IPLD schemas (protobuf, graphql, fuchsia FIDL) and gathering notes
+     - did you know?  basically nobody allows non-string map keys
+     - did you know?  graphql actually doesn't support maps at all (just structs)
+   - some experimental code for internal reprs of a type system.  just exploring.
+   - reading up on alternative chunkers.  need to digest more.  (ex; adler32 vs buzhash; casync switched from former to latter: why?)
+ - Blocked:
+   - more concrete IPLD data model specs is still on the wishlist.
+ - Next:
+   - need to do some PR reviewing (ex: ipld dagwalker)
+   - figure out where to land some of these notes and design sketches for IPLD
+  
+@travisperson
+ - Done:
+     - first pass at updating iptb
+ - Blocked: N / A
+ - Next:
+     - Mauppdate iptb with a few fixes
+     - update iptb to match ipfs gx deps
+     - release iptb 2.0
+  
+@aschmahmann
+ - Done:
+   - Basic ID-based message passing/protocol structure
+  - Research on GraphSync and planning next steps for graph synchronization related to multi-writer files
```

Comment:
@aschmahmann Are you aware of the new GraphSync specs? Read through https://github.com/ipld/specs/pulls

Reply:
I read about GraphSync a week or so ago, but I'll check out the latest.

Reply:
@diasdavid It's hard for me to tell how similar my "graph synchronization" issue is to GraphSync. My understanding of GraphSync is that it's designed to answer questions (or give the data corresponding to) the nodes descending from a Merkle root. The question I'm trying to answer is that if user A has a graph of changes (i.e. file versions where branching is allowed) and user B as another graph of changes, how can they best communicate so that they both get the graph Union(A,B). 

These problems are certainly close together, but I'm not sure the same implementation will address both issues. Any thoughts you or anyone else has are welcome though. Who working on GraphSync should I be speaking with?

Reply:
It seems that what you are looking for is something very close to the `coSelectors` thinking -- https://github.com/ipfs/notes/issues/154 (if Selectors are recursive, merkle root to leafs, coSelectors are merkleRoot coRecursive, pointing at future notes that point to our node).

Today, the way you do that is by having an append-only log that projects any other data structure that is the list of changes.

Reply:
This definitely shares similarities with the `coSelectors` you linked to. Just like you would have to do with synchronizing two append-only logs v1 of graph synchronization sends the entire graph between the two parties. The hope is that a later version of this type of graph synchronization requires less bandwidth per sync.

Should I try and leave this area mostly alone and focus on the interfaces and tooling that utilizes the simpler v1 implementation?